src/raft_log.c

```diff
@@ -233,25 +239,29 @@ int log_append(log_t* me_, raft_entry_t* ety, int *n)
 int log_delete(log_t* me_, int idx)
 {
     log_private_t* me = (log_private_t*)me_;
+    int e = 0;
 
-    if (!has_idx(me, idx))
+    if (0 == me->count || !has_idx(me, idx))
         return -1;
 
-    for (; idx <= me->base + me->count && me->count;)
+    while (me->base + 1 + me->count > idx)
```

Comment:
I think following the general style of this library we should write "idx <= me->base + me->count". It is also slightly easier to understand then "idx < me->base + me->count + 1", no?